packages/eslint-plugin-template/src/utils/get-original-attribute-name.ts

```diff
@@ -0,0 +1,63 @@
+import type {
+  TmplAstBoundAttribute,
+  TmplAstBoundEvent,
+  TmplAstTextAttribute,
+} from '@angular/compiler';
+import { BindingType, ParsedEventType } from '@angular/compiler';
+
+// TODO: It seems like `TmplAstBoundAttribute.type` is incorrectly typed at https://github.com/angular/angular/blob/2d1347b2ceef8c1e564c0f7cfe3a0a740b915e80/packages/compiler/src/render3/r3_ast.ts#L64.
+type BoundAttribute = Omit<TmplAstBoundAttribute, 'type'> &
+  Readonly<{
+    __originalType: BindingType;
+    type: 'BoundAttribute';
+  }>;
+
+// TODO: It seems like `TmplAstBoundEvent.type` is incorrectly typed at https://github.com/angular/angular/blob/2d1347b2ceef8c1e564c0f7cfe3a0a740b915e80/packages/compiler/src/render3/r3_ast.ts#L87.
+type BoundEvent = Omit<TmplAstBoundEvent, 'type'> &
+  Readonly<{
+    __originalType: ParsedEventType;
+    type: 'BoundEvent';
+  }>;
+
+/**
+ * Returns the original attribute name, respecting its `__originalType`.
+ * @example
+ * ```html
+ * <!-- The AST returns the name as "display", but we want the original name: "style.display.none" -->
+ * <div [style.display.none]="test"></div>
+ * ```
+ */
```

Comment:
I'm not good on documenting function so please suggest a better one if you want. (Here and elsewhere).